pkg/query/endpointset.go

```diff
@@ -183,28 +186,42 @@ type endpointSetNodeCollector struct {
 	storePerExtLset map[string]int
 
 	connectionsDesc *prometheus.Desc
+	labels          []string
 }
 
-func newEndpointSetNodeCollector() *endpointSetNodeCollector {
+func newEndpointSetNodeCollector(labels ...string) *endpointSetNodeCollector {
+	if len(labels) == 0 {
+		labels = []string{"external_labels", "store_type"}
+	}
 	return &endpointSetNodeCollector{
 		storeNodes: map[component.Component]map[string]int{},
 		connectionsDesc: prometheus.NewDesc(
 			"thanos_store_nodes_grpc_connections",
 			"Number of gRPC connection to Store APIs. Opened connection means healthy store APIs available for Querier.",
-			[]string{"external_labels", "store_type"}, nil,
+			labels, nil,
 		),
+		labels: labels,
 	}
 }
 
+func truncateExtLabels(s string, threshold int) string {
+	s = strings.Trim(s, "{}")
+	if len(s) > threshold {
+		s = s[:threshold]
+	}
+	return fmt.Sprintf("{%s}", s)
+}
+
```

Comment:
I think it can be still improved:

1. No need to trim left bracket
2. No need to trim before checking threshold - it's something we don't need to on the hot path.

```suggestion
// truncateExtLabels truncates the stringify external labels with the format of {labels..}.
func truncateExtLabels(s string, threshold int) string {
	if len(s) > threshold {
		return fmt.Sprintf("%s}", s[:threshold-1])
	}
	return s
}
```

Reply:
Oh I missed this

Reply:
looks cleaner